libbpf-tools/syscall_helpers.c

```diff
@@ -59,8 +59,8 @@ void init_syscall_names(void)
 		goto close;
 	}
 
-	/* skip the header */
-	fgets(buf, sizeof(buf), f);
+	/* skip the header, ignore the result of fgets, outwit the comiler */
+	(void) !!fgets(buf, sizeof(buf), f);
 
 	while (fgets(buf, sizeof(buf), f)) {
```

Comment:
Instead of outwitting the compiler, you can drop down to the error handling below.
```suggestion
	/* skip the header */
	char *p = fgets(buf, sizeof(buf), f);

	while (p && fgets(buf, sizeof(buf), f)) {
```

Reply:
Yes, but I personally prefer a `void` cast to a dangling unused pointer. We are not ignoring errors from `fgets`, because it will return another `NULL` in case of error right away.

In any case, the right solution is to add a parameter to the `ausyscall` utility to not print header, and eventually fix code here.